fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/FallbackCursor.java

```diff
@@ -31,48 +31,57 @@
 import javax.annotation.Nullable;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
-import java.util.function.Supplier;
+import java.util.function.Function;
 
 /**
  * Provide an alternative cursor in case the primary cursor fails. This cursor has an <code>inner</code> cursor that is
  * used to return the results in the sunny day scenarios, and an alternative provider of a <code>fallback</code> cursor
  * that will be used if the primary cursor encounters an error.
- *
- * Note that there are business rules around when a fallback may happen. For example, because of the way records are
- * returned to the consumer, once a record is returned from the primary cursor, no fallback may be permitted: This is
- * done in order to prevent the case where a few records are returned, then a failure happens and the fallback cursor
- * starts again from the beginning, resulting in duplicate records being returned.
- * In practice, since many errors are observed when the request is sent to FDB (which coincide with the cursor's
- * first <code>onNext()</code> call, many such failures will be caught by that first result.
- *
+ * In order to continue from a failed iteration, the cursor will pass the last successful result to the provider
+ * of the fallback cursor. <b>It is the responsibility</b> of the fallback cursor provider to ensure that it can resume
+ * the iteration from that point. Among other things, that it can create an iteration over the range that starts with
```

Comment:
```suggestion
 * the iteration from that point. Among other things, this means starting a scan over the range that starts with
```

This is maybe a little clearer (arguably)